### PR TITLE
Polish Azure.ResourceManager.Automation for GA

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs [hunk 1]
```diff
@@ -1601,14 +1601,14 @@ public virtual Response<LinkedWorkspace> GetLinkedWorkspace(CancellationToken ca
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentException"> <paramref name="typeName"/> is an empty string, and was expected to be non-empty. </exception>
         /// <exception cref="ArgumentNullException"> <paramref name="typeName"/> is null. </exception>
-        /// <returns> An async collection of <see cref="TypeField" /> that may take multiple service requests to iterate over. </returns>
-        public virtual AsyncPageable<TypeField> GetFieldsByTypeObjectDataTypesAsync(string typeName, CancellationToken cancellationToken = default)
+        /// <returns> An async collection of <see cref="AutomationModuleField" /> that may take multiple service requests to iterate over. </returns>
+        public virtual AsyncPageable<AutomationModuleField> GetFieldsByTypeAsync(string typeName, CancellationToken cancellationToken = default)
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs:1605]: Maybe `GetFieldsByObjectDataType` is a better method name.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationJobScheduleData.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            Runbook = runbook;
+            RunOn = runOn;
+            Parameters = parameters;
+        }
+
+        /// <summary>
+        /// Gets or sets the id of job schedule.
+        /// Serialized Name: JobSchedule.properties.jobScheduleId
+        /// </summary>
+        public string JobScheduleId { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationJobScheduleData.cs:62]: Is this a ResourceIdentifier?

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/RunbookTypeEnum.cs [hunk 3]
```diff
@@ -10,7 +10,10 @@
 
 namespace Azure.ResourceManager.Automation.Models
 {
-    /// <summary> Gets or sets the type of the runbook. </summary>
+    /// <summary>
+    /// Gets or sets the type of the runbook.
+    /// Serialized Name: RunbookTypeEnum
+    /// </summary>
     public readonly partial struct RunbookTypeEnum : IEquatable<RunbookTypeEnum>
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/RunbookTypeEnum.cs:17]: I don't think `Enum` is a good suffix here, maybe we should drop it.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountPython2PackageResource.cs [hunk 4]
```diff
@@ -35,7 +35,7 @@ public static ResourceIdentifier CreateResourceIdentifier(string subscriptionId,
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountPython2PackageResource.cs:35]: Should we use digit in the resource name? It's a little strange.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationActivityParameter.cs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.Automation.Models
+{
+    /// <summary>
+    /// Definition of the activity parameter.
+    /// Serialized Name: ActivityParameter
+    /// </summary>
+    public partial class AutomationActivityParameter
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationActivityParameter.cs:17]: Maybe we should rename it to `AutomationActivityParameterDefinition`.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            WeekDays = weekDays;
+            MonthDays = monthDays;
+            MonthlyOccurrences = monthlyOccurrences;
+        }
+
+        /// <summary>
+        /// Days of the week that the job should execute on.
+        /// Serialized Name: AdvancedSchedule.weekDays
+        /// </summary>
+        public IList<string> WeekDays { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs:51]: `DaysOfWeek` looks better.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        /// <summary>
+        /// Days of the month that the job should execute on. Must be between 1 and 31.
+        /// Serialized Name: AdvancedSchedule.monthDays
+        /// </summary>
+        public IList<int> MonthDays { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs:56]: `DaysOfMonth` looks better.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationCountType.cs [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        /// <summary>
+        /// status
+        /// Serialized Name: CountType.status
+        /// </summary>
+        public static AutomationCountType Status { get; } = new AutomationCountType(StatusValue);
+        /// <summary>
+        /// nodeconfiguration
+        /// Serialized Name: CountType.nodeconfiguration
+        /// </summary>
+        public static AutomationCountType Nodeconfiguration { get; } = new AutomationCountType(NodeconfigurationValue);
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationCountType.cs:40]:
```suggestion
        public static AutomationCountType NodeConfiguration { get; } = new AutomationCountType(NodeconfigurationValue);
```

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationErrorResponse.cs [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+#nullable disable
+
+namespace Azure.ResourceManager.Automation.Models
+{
+    /// <summary>
+    /// Error response of an operation failure
+    /// Serialized Name: ErrorResponse
+    /// </summary>
+    public partial class AutomationErrorResponse
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationErrorResponse.cs:14]: In Azure.Core we defined the common type `ResponseError`, so here I think we should rename this one to `AutomationResponseError` to keep consistent.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationModuleField.cs [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+        /// <summary>
+        /// Gets or sets the name of the field.
+        /// Serialized Name: TypeField.name
+        /// </summary>
+        public string Name { get; }
+        /// <summary>
+        /// Gets or sets the type of the field.
+        /// Serialized Name: TypeField.type
+        /// </summary>
+        public string TypeFieldType { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationModuleField.cs:45]: Bad name, suggest renaming to `FieldType`

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/LinuxUpdateClass.cs [hunk 11]
```diff
@@ -10,7 +10,10 @@
 
 namespace Azure.ResourceManager.Automation.Models
 {
-    /// <summary> Update classifications included in the software update configuration. </summary>
+    /// <summary>
+    /// Update classifications included in the software update configuration.
+    /// Serialized Name: LinuxUpdateClasses
+    /// </summary>
     public readonly partial struct LinuxUpdateClass : IEquatable<LinuxUpdateClass>
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/LinuxUpdateClass.cs:17]: Suggest renaming to `LinuxUpdateClassification`

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateClass.cs [hunk 12]
```diff
@@ -10,7 +10,10 @@
 
 namespace Azure.ResourceManager.Automation.Models
 {
-    /// <summary> Update classification included in the software update configuration. A comma separated string with required values. </summary>
+    /// <summary>
+    /// Update classification included in the software update configuration. A comma separated string with required values
+    /// Serialized Name: WindowsUpdateClasses
+    /// </summary>
     public readonly partial struct WindowsUpdateClass : IEquatable<WindowsUpdateClass>
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateClass.cs:17]: Suggest renaming to `WindowsUpdateClassification`

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+        /// <summary>
+        /// Update classification included in the software update configuration. A comma separated string with required values
+        /// Serialized Name: WindowsProperties.includedUpdateClassifications
+        /// </summary>
+        public WindowsUpdateClass? IncludedUpdateClassifications { get; set; }
+        /// <summary>
+        /// KB numbers excluded from the software update configuration.
+        /// Serialized Name: WindowsProperties.excludedKbNumbers
+        /// </summary>
+        public IList<string> ExcludedKbNumbers { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs:60]: `KB`?

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+        /// <summary>
+        /// KB numbers included from the software update configuration.
+        /// Serialized Name: WindowsProperties.includedKbNumbers
+        /// </summary>
+        public IList<string> IncludedKbNumbers { get; }
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs:65]: `KB`?

File: sdk/automation/Azure.ResourceManager.Automation/src/autorest.md [hunk 15]
```diff
@@ -15,16 +15,173 @@ skip-csproj: true
 modelerfour:
   flatten-payloads: false
 
+mgmt-debug:
```
Comment (Yao725) [inline sdk/automation/Azure.ResourceManager.Automation/src/autorest.md:18]: Maybe we should remove this configuration after debug.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountModuleResource.cs [hunk 16]
```diff
@@ -399,14 +399,14 @@ Page<Activity> NextPageFunc(string nextLink, int? pageSizeHint)
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentException"> <paramref name="typeName"/> is an empty string, and was expected to be non-empty. </exception>
         /// <exception cref="ArgumentNullException"> <paramref name="typeName"/> is null. </exception>
-        /// <returns> An async collection of <see cref="TypeField" /> that may take multiple service requests to iterate over. </returns>
-        public virtual AsyncPageable<TypeField> GetFieldsByModuleAndTypeObjectDataTypesAsync(string typeName, CancellationToken cancellationToken = default)
+        /// <returns> An async collection of <see cref="AutomationModuleField" /> that may take multiple service requests to iterate over. </returns>
+        public virtual AsyncPageable<AutomationModuleField> GetFieldsByModuleAndTypeAsync(string typeName, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountModuleResource.cs:403]: rename to `GetFieldsByObjectDataType`

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs [hunk 17]
```diff
@@ -1510,9 +1510,9 @@ Page<AutomationUsage> FirstPageFunc(int? pageSizeHint)
         /// Operation Id: Keys_ListByAutomationAccount
         /// </summary>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual async Task<Response<KeyListResult>> GetByAutomationAccountKeyAsync(CancellationToken cancellationToken = default)
+        public virtual async Task<Response<AutomationKeyListResult>> GetAutomationAccountKeysAsync(CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs:1513]: This is a post get method, which is different.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs:1605]: This is a shared function for both `AutomationAccountResource` and `AutomationAccountModuleResource`, that in `AutomationAccountModuleResource` there is another method return the same value but different scope, one for module one for account.
Related to your first comment, so leave it as is here.

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs (hunk 17 above)
[the remark this answers is not in the document]
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationAccountResource.cs:1513]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationJobScheduleData.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/AutomationJobScheduleData.cs:62]: uuid, fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/RunbookTypeEnum.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/RunbookTypeEnum.cs:17]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationActivityParameter.cs (hunk 5 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationActivityParameter.cs:17]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs:51]: WeekDays is the right meaning here

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs (hunk 7 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationAdvancedSchedule.cs:56]: MonthDays is the right meaning here

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationCountType.cs (hunk 8 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationCountType.cs:40]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationErrorResponse.cs (hunk 9 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationErrorResponse.cs:14]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationModuleField.cs (hunk 10 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/AutomationModuleField.cs:45]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/LinuxUpdateClass.cs (hunk 11 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/LinuxUpdateClass.cs:17]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateClass.cs (hunk 12 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateClass.cs:17]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs (hunk 13 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs:60]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs (hunk 14 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/Generated/Models/WindowsUpdateConfigurationProperties.cs:65]: fixed

File: sdk/automation/Azure.ResourceManager.Automation/src/autorest.md (hunk 15 above)
Comment (ArthurMa1978) [reply sdk/automation/Azure.ResourceManager.Automation/src/autorest.md:18]: removed

**Review by Yao725 — APPROVED** [review]

**Pull request merged**